From 8eae21743b1dd8b4f83feaaa115dae6046e65bf9 Mon Sep 17 00:00:00 2001
Message-Id: <8eae21743b1dd8b4f83feaaa115dae6046e65bf9.1626397765.git.bobby.eshleman@gmail.com>
From: Bobby Eshleman <bobby.eshleman@gmail.com>
Date: Wed, 23 Jun 2021 11:15:45 -0700
Subject: [PATCH] varstore-sb-state: only load auth data if needed

The loaded auth data is only necessary when calling varstore-sb-state
user, so only call load_one_auth_data() if varstore-sb-state user is
called.

This avoids varstore-sb-state ${uuid} setup from attempting to load auth
files, which it doesn't actually need, and therefore avoids unnecessary
complaints if there happens to be one or more missing auth files.

Signed-off-by: Bobby Eshleman <bobby.eshleman@gmail.com>
---
 tools/varstore-sb-state.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/varstore-sb-state.c b/tools/varstore-sb-state.c
index dc91b2b..6305dd2 100644
--- a/tools/varstore-sb-state.c
+++ b/tools/varstore-sb-state.c
@@ -92,7 +92,8 @@ int main(int argc, char **argv)
     if (opt_socket)
         db->parse_arg("socket", opt_socket);
 
-    load_auth_data();
+    if (!strcmp(argv[optind + 1], "user"))
+        load_auth_data();
 
     if (!drop_privileges(opt_chroot, opt_depriv, opt_gid, opt_uid))
         exit(1);
-- 
2.30.0

